src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1028,7 +1028,11 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
 
             var typeInfo = model.GetTypeInfo(decl);
             Assert.Equal(expectedType, typeInfo.Type);
-            Assert.Equal(expectedType, typeInfo.ConvertedType);
+            if ((expectedSymbol as LocalSymbol)?.DeclarationKind == LocalDeclarationKind.OutVariable)
```

Comment:
I'll change this so it handles fields too. However, I'm not intending to address https://github.com/dotnet/roslyn/issues/17377 (regarding misplaced declaration expressions) in the context of this PR.